lib/spack/spack/solver/asp.py

```diff
@@ -3968,6 +3976,31 @@ def _specs_from_mirror():
         return []
 
 
+def _specs_from_environment(path):
+    active_env = ev.active_environment()
+
+    if path:
+        if not ev.is_env_dir(path) and ev.exists(path):
+            path = ev.root(path)
+
+    if active_env and path in active_env.included_specs_by_hash:
```

Comment:
I believe this makes sense as the desired behavior.

If we reuse an environment path explicitly:
* And that environment path is included concretely, then reuse the included concrete specs
* And that path is *not* included concretely, then reuse from the concrete environment at that path